src/main.rs

```diff
@@ -87,9 +87,30 @@ impl std::str::FromStr for Color {
     }
 }
 
+fn get_version() -> String {
+    let pkg_version = if built_info::CI_PLATFORM.is_some() && built_info::PKG_VERSION.contains('-')
+    {
+        built_info::PKG_VERSION.replace("-dev", "-nightly")
```

Comment:
This gets into versioning questions:

1. Are we planning on representing the next version with the `-dev` suffix? If so, how do we know which version that is? For example, is the next version `0.6.0` or `1.0.0`?
2. Or, should we forgo guessing the next version and allow the package version to represent the last released version? We could also use the result of `git describe --tags` here. I believe that is `built_info::GIT_VERSION`.

We started with this discussion but punted on a conclusion since it involved figuring out how we'll release point releases.

Reply:
In my ideal world, we use the version of the next release + `-dev`. But since that requires knowing the future, I think my second choice would be most recent release + `-dev`?

Reply:
I think that if we go with "the next version in Cargo.toml" approach, there might even be no need to use `-dev` suffix, it could be just the next version.

* Rust sets version in `master` to the next released version, without any suffixes ([link](https://github.com/rust-lang/rust/commits/master/src/bootstrap/channel.rs))
* Clang does the same, as `clang --version` from master shows the next version to be released

Then the question is it an actual release or not could be decided in CI by looking at presence of the corresponding Git tag. For example, if the version in `Cargo.toml` is 0.6.0 and there is no Git tag `v0.6.0`, then the version output by `vector --version` would be `vector 0.6.0-dev (g816041c 2019-10-17)` and if there is tag `v0.6.0` then `vector --version` would return `vector 0.6.0 (g816041c 2019-10-17)`.

The case when pre-release version is distributed as an archive is rare and I think that there is no need to officially distribute source archives of master at all, it should be enough to distribute only release archives.

Reply:
Well said, and I agree. There is nothing stopping up from bumping up to a major version when it warrants it. My vote is to adopt this strategy. Ref https://github.com/timberio/vector/issues/1010.